experimental/LieAlgebras/src/CoxeterGroup.jl

```diff
@@ -7,26 +7,31 @@ Return the Coxeter matrix $m$ associated to the Cartan matrix `gcm`. If there is
 then this will be expressed by $m_{ij} = 0$ (instead of the usual convention $m_{ij} = \infty$).
 The keyword argument `check` can be set to `false` to skip verification whether `gcm` is indeed a generalized Cartan matrix.
 """
-function coxeter_from_cartan_matrix(gcm; check::Bool=true)
+function coxeter_from_cartan_matrix(gcm::ZZMatrix; check::Bool=true)
   @req !check || is_cartan_matrix(gcm) "requires a generalized Cartan matrix"
 
-  cm = identity_matrix(gcm)
   rk, _ = size(gcm)
-  for i in 1:rk, j in 1:rk
-    if i == j
-      continue
-    end
+  cm = matrix(
+    ZZ, [coxeter_matrix_entry_from_cartan_matrix(gcm, i, j) for i in 1:rk, j in 1:rk]
+  )
 
-    if gcm[i, j] == 0
-      cm[i, j] = 2
-    elseif gcm[i, j] == -1
-      cm[i, j] = 3
-    elseif gcm[i, j] == -2
-      cm[i, j] = 4
-    elseif gcm[i, j] == -3
-      cm[i, j] = 6
-    end
+  return cm
+end
+
+function coxeter_matrix_entry_from_cartan_matrix(gcm::ZZMatrix, i::Int, j::Int)
+  if i == j
+    return 1
   end
 
-  return cm
+  if gcm[i, j] * gcm[j, i] == 0
+    return 2
+  elseif gcm[i, j] * gcm[j, i] == 1
+    return 3
+  elseif gcm[i, j] * gcm[j, i] == 2
+    return 4
+  elseif gcm[i, j] * gcm[j, i] == 3
```

Comment:
Maybe apply this micro optimization?
```suggestion
  d = gcm[i, j] * gcm[j, i]
  if d == 0
    return 2
  elseif d == 1
    return 3
  elseif d == 2
    return 4
  elseif d == 3
```